test/test-shell.sh

```diff
@@ -35,6 +35,7 @@ for i in $DIR/test/shell/*.sh; do
 	mkdir -p '/tmp/mgmt/'	# directory for mgmt to put files in
 	#echo "Running: $ii"
 	export MGMT_TMPDIR='/tmp/mgmt/'	# we can add to env like this
+	chmod 777 $MGMT_TMPDIR # we need other users to be able to write in it
```

Comment:
I only looked at exec-usergroup.sh and the test itself looks good. You might want to think about putting the test script in a subfolder like /tmp/mgmt/exec-usergroup-test/
Also, 777 is usually not a good idea security-wise. Check out sticky bits for a possible solution.
https://en.wikipedia.org/wiki/Sticky_bit